test/jasmine/tests/plots_test.js

```diff
@@ -54,7 +54,7 @@ describe('Test Plots', function() {
             expect(gd._fullData[1].z).toBe(newData[1].z);
             expect(gd._fullData[1]._empties).toBe(oldFullData[1]._empties);
             expect(gd._fullLayout.scene._scene).toBe(oldFullLayout.scene._scene);
-            expect(gd._fullLayout._plots).toBe(oldFullLayout._plots);
+            expect(gd._fullLayout._plots.plot).toBe(oldFullLayout._plots.plot);
```

Comment:
So it looks like it reconstructs a *new* `_plots` object with the same contents vs. just transferring it? Is that what makes the join work correctly?